src/main/java/de/tum/in/www1/artemis/service/CourseService.java

```diff
@@ -411,7 +411,8 @@ public List<StudentDTO> registerUsersForCourseGroup(Long courseId, List<StudentD
         for (var studentDto : studentDTOs) {
             var registrationNumber = studentDto.getRegistrationNumber();
             var login = studentDto.getLogin();
-            Optional<User> optionalStudent = userService.findUserAndAddToCourse(registrationNumber, courseGroupName, courseGroupRole, login);
+            var email = studentDto.getEmail();
```

Comment:
💡The above Doc for this method seems to be outdated now:
```
     * Note: registration based on other user attributes (e.g. email, name, login) is currently NOT supported
```
Maybe you could adapt it?